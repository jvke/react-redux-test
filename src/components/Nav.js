import React from 'react';
import styled from 'styled-components';
import { transparentize } from 'polished';
import { Link } from 'react-router-dom';

const Nav = (props) => (
  <div className={props.className}>
    <Link className='logo' to='/'>GitSearch</Link>
    <div className="nav-right">
      <a>D3 demo (coming soon)</a>
    </div>
  </div>
);

export default styled(Nav)`
  overflow: hidden;
  padding: 15px 10px;

  .logo {
    color: ${props => props.theme.textColor};
    text-decoration: none;
    font-size: 24px;
    font-family: 'Source Serif Pro', serif;
  }

  .nav-right {
    float: right;

    & a {
      padding: 0px 10px;
      color: ${props => transparentize(0.2, props.theme.textColor)}
    }
  }
`;
