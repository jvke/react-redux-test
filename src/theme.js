export default {
  brand: {
    light: 'rgb(8, 62, 101)',
    dark: 'rgb(1, 20, 46)',
  },
  alert: {
    error: '#FF0000'
  },
  textColor: '#EEEEEE'
}
