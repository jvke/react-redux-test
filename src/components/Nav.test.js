import React from 'react';
import ReactDOM from 'react-dom';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import Nav from './Nav';
import theme from '../theme';
import { MemoryRouter } from 'react-router-dom';

describe('Nav', () => {
  it('Nav renders with no props specified', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Nav theme={theme} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

})
