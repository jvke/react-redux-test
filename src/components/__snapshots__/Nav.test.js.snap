// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav renders with no props specified 1`] = `
.c0 {
  overflow: hidden;
  padding: 15px 10px;
}

.c0 .logo {
  color: #EEEEEE;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 24px;
  font-family: 'Source Serif Pro',serif;
}

.c0 .nav-right {
  float: right;
}

.c0 .nav-right a {
  padding: 0px 10px;
  color: rgba(238,238,238,0.8);
}

<div
  className="c0"
>
  <a
    className="logo"
    href="/"
    onClick={[Function]}
  >
    GitSearch
  </a>
  <div
    className="nav-right"
  >
    <a>
      D3 demo (coming soon)
    </a>
  </div>
</div>
`;
